x-pack/plugins/monitoring/server/lib/alerts/fetch_index_shard_size.ts

```diff
@@ -143,9 +137,21 @@ export async function fetchIndexShardSize(
         _source: { source_node: sourceNode, index_stats: indexStats },
       } = topHit;
 
-      const { size_in_bytes: shardSizeBytes } = indexStats?.primaries?.store!;
+      if (!indexStats || !indexStats.primaries) {
+        continue;
+      }
+
+      const { primaries: totalPrimaryShards } = indexStats.shards;
+      const { size_in_bytes: primaryShardSizeBytes = 0 } = indexStats.primaries.store!;
+      if (!primaryShardSizeBytes) {
+        continue;
+      }
       const { name: nodeName, uuid: nodeId } = sourceNode;
-      const shardSize = +(shardSizeBytes! / gbMultiplier).toFixed(2);
+      const avgShardSize = primaryShardSizeBytes ? primaryShardSizeBytes / totalPrimaryShards : 0;
```

Comment:
We're already defaulting this value to `0` on line 145, right? `0 / totalPrimaryShards` will be `0` so do we need this ternary?

Also, do we ever expect `totalPrimaryShards` to be 0?

Reply:
Right, the ternary actually checks to make sure it's not 0 (which can be either from the response or our default), since I don't want to divide the 0 (like you mentioned). Maybe scoping would make it more readable? eg:
```javascript
avgShardSize = primaryShardSizeBytes ? (primaryShardSizeBytes / totalPrimaryShards) : 0
```

> ...do we ever expect totalPrimaryShards to be 0

Yes, it's possible to have zero primaries (mainly during the allocation of shards), in which case the cluster status goes red (though in most cases it's temporary since not all shards are yet assigned).  [This](https://www.elastic.co/blog/red-elasticsearch-cluster-panic-no-longer) article explains it very well

Reply:
I guess I don't even need this check, since I'm already doing this right before:
```javascript
if (!primaryShardSizeBytes) {
  continue;
}
```

Reply:
Exactly, that's what I was thinking. I asked about `totalPrimaryShards` because we're currently not checking that value to make sure it's not 0, so I think we can still end up in a divide by 0 scenario here, right?

Reply:
No, because the condition  `if (primaryShardSizeBytes)` would not pass because 0 is actually considered falsy in javascript/ts. So, we don't need to explicitly say  `if (primaryShardSizeBytes === 0)`

Reply:
Heh, yeah I know about 0 being falsy ;) I'm asking about the denominator, `totalPrimaryShards`.

Reply:
Eh, I see your point now, sorry for the derp moment 🙃 

Added that to the check as well, so should be good now